modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardPanel.ts

```diff
@@ -598,7 +598,17 @@ export class ContentWizardPanel
         return new ContentWizardToolbar({
             actions: this.wizardActions,
             workflowStateIconsManager: this.workflowStateManager,
-            className: 'content-wizard-toolbar'
+            className: 'content-wizard-toolbar',
+            compareVersionsPreHook: () => {
+                if (!this.hasUnsavedChanges()) {
+                    return Q.resolve();
+                }
+
+                const deferred: Q.Deferred<void> = Q.defer<void>();
+                this.saveChangesWithoutValidation().then(() => deferred.resolve());
+
+                return deferred.promise;
```

Comment:
Can be simplified to:
```
return this.saveChangesWithoutValidation().then(() => null);
```

Reply:
No, it's not the same. The prehook method returns a promise which gets resolved after save. Simply returning null won't resolve the promise.

Reply:
Returning simple value or throwing an exception inside a promise resolves that promise so that will work ;)

Reply:
Changed return type to `Q<any>` and changed the hook body to:
```
compareVersionsPreHook: () => {
    if (!this.hasUnsavedChanges()) {
        return Q.resolve();
    }

    return this.saveChangesWithoutValidation();
}
```